### PR TITLE
PMM-3832 MySQL EXPLAIN action API.

### PR DESCRIPTION
**Pull request by DexterHD**

### CODE REVIEW

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-362/

File: api/managementpb/actions.proto [hunk 1]
```diff
@@ -93,9 +93,10 @@ message StartMySQLExplainActionRequest {
   string pmm_agent_id = 1;
   // Service ID for this Action.
   string service_id = 2;
-  //
-  // TODO specific parameters if we need them.
-
+  // database for query.
+  string db = 3;
+  // query example.
+  string query = 4;
```
Comment (BupycHuk) [inline api/managementpb/actions.proto]: I think these params aren't enough. 
There are no credentials (username and password) to prepare dsn.

Comment (BupycHuk) [reply api/managementpb/actions.proto]: And in StartMySQLJsonExplainActionRequest as well

Comment (DexterHD) [reply api/managementpb/actions.proto]: Do you mean we can't get credentials from service and agent?

Comment (DexterHD) [reply api/managementpb/actions.proto]: I'm not sure about `db` parameter on UI but it explained in old implementation:
> // If the query has a default db, use it; else, all tables need to be db-qualified
> // or EXPLAIN will throw an error.

Comment (BupycHuk) [reply api/managementpb/actions.proto]: I mean there is no agent_id param to get credentials from mysqld_exporter and no username and password params to use them. 
are you going to use some logic inside of pmm-managed to get credentials from mysqld_exporter for current service? like `findAgentForService(MYSQLD_EXPORTER, req.ServiceID)`?

Comment (DexterHD) [reply api/managementpb/actions.proto]: > are you going to use some logic inside of pmm-managed to get credentials from mysqld_exporter for current service? like findAgentForService(MYSQLD_EXPORTER, req.ServiceID)?
It seems to. We haven't requirements and didn't discuss this. 

@AlekSi @rnovikovP Need your help with that. What should I do?

Comment (AlekSi) [reply api/managementpb/actions.proto]: Before GA we should add default username/password to Service itself. They should be used by Agents and Actions when explicit username/password are not given. It is currently tracked there: https://jira.percona.com/browse/PMM-3786, but will be extracted into separate one a bit later.

In the meantime, let's do `findAgentForService(MYSQLD_EXPORTER, req.ServiceID)` with a comment indicating that is a temporary hack.

File: api/agentpb/agent.proto [hunk 2]
```diff
@@ -91,8 +98,10 @@ message StartActionRequest {
   }
   // MySQLExplainParams describes MySQL Explain action parameters.
   message MySQLExplainParams {
-    // TODO
-
+    string dsn = 1;
+    string db = 2;
```
Comment (AlekSi) [inline api/agentpb/agent.proto]: Isn't database name a part of DSN?

File: api/managementpb/actions.proto [hunk 3]
```diff
@@ -105,6 +106,24 @@ message StartMySQLExplainActionResponse {
   string pmm_agent_id = 2;
 }
 
+message StartMySQLJsonExplainActionRequest {
```
Comment (AlekSi) [inline api/managementpb/actions.proto]: StartMySQLExplainJSONActionRequest. Common prefix "StartMySQLExplain" for grouping, JSON for acronym.

File: api/agentpb/agent.proto (hunk 2 above)
Comment (DexterHD) [reply api/agentpb/agent.proto]: Yes, but I'm not sure it enough:

https://github.com/percona/qan-agent/blob/9399e1fe46df02e988838816c0578282918b0b8c/query/plugin/mysql/explain/explain.go#L62-L69

Comment (AlekSi) [reply api/agentpb/agent.proto]: We will extract it from DSN.
If we have two options to pass database name, we will have 4 combinations we will need to handle. I don't think we should.

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-362/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-362/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-362/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-362/

**Pull request merged**